scripts/db-heal-test.sh

```diff
@@ -0,0 +1,127 @@
+################
+#--- Script ---#
+################
+
+# Run the command in the background and redirect stdout and stderr to a log file
+log_file="$(pwd)/output.log"
+current=$(pwd)
+
+# First iteration has different command
+cmd="./build/aida-vm-sdb substate --validate --db-tmp "$tmp_path" --carmen-schema 5 --db-impl carmen --aida-db "$aida_db_path" --no-heartbeat-logging --track-progress --archive --archive-variant s5 --archive-query-rate 200 --carmen-cp-interval 200 "$first_block" "$last_block""
+cd $aida_path
+$cmd &> "$log_file" &
+command_pid=$!
+cd $current
+
+echo "Creating database with aida-vm-sdb..."
+
+# Function to monitor the log file
+monitor_log() {
+  tail -F "$log_file" | while read LINE; do
+    echo "$LINE" | grep -q "block $kill_block"
+    if [ $? -eq 0 ]; then
+      echo "Interrupting."
+      kill $command_pid
+      exit 0
+    fi
+  done
+}
+
+# Start monitoring the log file
+monitor_log
+
+# Wait for the first command to complete
+wait $command_pid
+
+# Find working dir
+working_dir=$(ls -td "$tmp_path"/*/ | head -1)
+archive="${working_dir}archive"
+live="${working_dir}live"
+
+echo "Testing db created, starting loop."
+
+# Save last working dir to remove it
+last_working_dir=""
+
+for ((i=1; i<=number_of_iterations; i++)); do
+  # Restore Archive
+  (cd $carmen_root && go run ./database/mpt/tool reset --force-unlock "$archive" "$restore_block")
+
+  # Export genesis to restore LiveDB
+  genesis="${tmp_path}test_genesis.dat"
+
+  echo "Restoration complete. Exporting LiveDB genesis block "$restore_block"."
+  (cd $carmen_root && go run ./database/mpt/tool export --block "$restore_block" "$archive" "$genesis")
+
+  # Restore LiveDB
+  echo "Export complete. Applying LiveDB genesis."
+  rm -rf "$live"
+  (cd $carmen_root && go run ./database/mpt/tool import-live-db "$genesis" "$live")
+
+  echo "Iteration "$i"/"$number_of_iterations""
+  first_block=$((restore_block + 1))
+  last_block=$((last_block + 1000))
+  restore_block=$((restore_block + 1000))
+  kill_block=$((kill_block + 1000))
+
+  echo "Syncing to block "$last_block"..."
+  command="./build/aida-vm-sdb substate --validate --db-tmp "$tmp_path" --carmen-schema 5 --db-impl carmen --aida-db "$aida_db_path" --no-heartbeat-logging --track-progress --archive --archive-variant s5 --archive-query-rate 200 --carmen-cp-interval 200 --db-src "$working_dir" --skip-priming "$first_block" "$last_block""
+
+  cd $aida_path
+  $command &> "$log_file" &
+  command_pid=$!
+  cd $current
+
+  # Start monitoring the log file
+  monitor_log
+
+  # Wait for the command to complete
+  wait $command_pid
+
+  # Remove unused dir and save current dir to be deleted next iteration.
+  echo "Removing previous database"$last_working_dir""
+  rm -rf "$last_working_dir"
+  last_working_dir="$working_dir"
+
+  # Re-find working dir - Aida copies db-src
+  working_dir=$(ls -td "$tmp_path"/*/ | head -1)
+  archive="${working_dir}archive"
+  live="${working_dir}live"
+
+done
+
+rm $log_file
+
```

Comment:
I guess all remaining tmp files should be deleted here. The last iteration was not deleted right?

Reply:
good catch